### ISSUE TITLE
Storing custom properties in a Network

### ISSUE THREAD

**Issue by bartvollebregt** [deeplearning4j/deeplearning4j#2735]
I'm creating and saving a MultiLayerNetwork to a file. I've some metadata (like last trained timestamp, and amount of predictions done) that I'd like to store in the Network or the Networks config as custom properties.
This way we are easily able to retrieve and use the metadata when restoring the Network

After discussing a bit with Melanie Warrick on Gitter we agreed it would be cool if this could be could be implemented so hereby the feature request.

**Comment by turambar** [comment]
This is related to some of our discussions around transfer learning @nyghtowl @eraly @AlexDBlack.

**Comment by raver119** [comment]
Technically, we could extend ModelSerializer to accept something like serializable Properties. And this would allow user to store arbitrary information in his model file.

**Comment by AlexDBlack** [comment]
Right, it could be a Properties object. Or, in principle we could easily provide a addObject(String key, Serializable) and getObject(String key, Class<?> type) methods for this - allow users to store just about anything they want. Probably stored in a subdirectory in the zip.

If we do that, I'd want it to be self describing: i.e., be able to list the stored (non-standard) data, along with the types.
